src/pages/api/assistant/trip.ts

```diff
@@ -0,0 +1,59 @@
+import { parseFilterQuery } from '@atb/components/transport-mode-filter/utils';
+import { errorResultAsJson, tryResult } from '@atb/modules/api-server';
+import { TripApiReturnType } from '@atb/page-modules/assistant/client/journey-planner';
+import { handlerWithAssistantClient } from '@atb/page-modules/assistant/server';
+import { parseTripQuery } from '@atb/page-modules/assistant/utils';
+import { ServerText } from '@atb/translations';
+import { constants } from 'http2';
+
+export default handlerWithAssistantClient<TripApiReturnType>({
+  async GET(req, res, { client, ok }) {
+    const tripQuery = parseTripQuery(req.query);
+    if (tripQuery) {
```

Comment:
Could reduce nesting with early exit here. I think this will make it easier to follow the code further down as you don't have to remember the context in the same way:

```suggestion
    if (!tripQuery) {
      return errorResultAsJson(
        res,
        constants.HTTP_STATUS_BAD_REQUEST,
        ServerText.Endpoints.invalidData,
      );
    }
```